contract_manager/scripts/deploy_evm_pulse_contracts.ts

```diff
@@ -0,0 +1,206 @@
+    fs.readFileSync(
+      path.join(config.jsonOutputDir, "PulseUpgradeable.json"),
+      "utf8"
+    )
+  );
+  console.log("PulseArtifact bytecode type:", typeof pulseArtifact.bytecode);
+
+  const pulseImplAddr = await deployIfNotCached(
+    CACHE_FILE,
+    chain,
+    config,
+    "PulseUpgradeable",
+    []
+  );
+
+  console.log("PulseUpgradeable implementation deployed at:", pulseImplAddr);
+
+  const pulseImplContract = getWeb3Contract(
+    config.jsonOutputDir,
+    "PulseUpgradeable",
+    pulseImplAddr
+  );
+
+  console.log("Preparing initialization data...");
+
+  const pulseInitData = pulseImplContract.methods
+    .initialize(
+      executorAddr, // owner
+      executorAddr, // admin
+      "1", // pythFeeInWei
+      executorAddr, // pythAddress - using executor as a placeholder
+      chain.isMainnet()
+        ? PULSE_DEFAULT_PROVIDER.mainnet
+        : PULSE_DEFAULT_PROVIDER.testnet,
+      true, // prefillRequestStorage
+      3600 // exclusivityPeriodSeconds - 1 hour
+    )
+    .encodeABI();
+
+  console.log("Deploying ERC1967Proxy for Pulse...");
+
+  return await deployIfNotCached(
+    CACHE_FILE,
+    chain,
+    config,
+    "ERC1967Proxy",
+    [pulseImplAddr, pulseInitData],
+    // NOTE: we are deploying a ERC1967Proxy when deploying executor
+    // we need to provide a different cache key. As the `artifactname`
+    // is same in both case which means the cache key will be same
+    `${chain.getId()}-ERC1967Proxy-PULSE1`
+  );
+}
+
+async function topupAccountsIfNecessary(
+  chain: EvmChain,
+  deploymentConfig: DeploymentConfig
+) {
+  for (const [accountName, defaultAddresses] of [
+    ["keeper", PULSE_DEFAULT_KEEPER],
+    ["provider", PULSE_DEFAULT_PROVIDER],
+  ] as const) {
+    const accountAddress = chain.isMainnet()
+      ? defaultAddresses.mainnet
+      : defaultAddresses.testnet;
+    const web3 = chain.getWeb3();
+    const balance = Number(
+      web3.utils.fromWei(await web3.eth.getBalance(accountAddress), "ether")
+    );
+    const MIN_BALANCE = 0.01;
+    console.log(`${accountName} balance: ${balance} ETH`);
+    if (balance < MIN_BALANCE) {
+      console.log(
+        `Balance is less than ${MIN_BALANCE}. Topping up the ${accountName} address...`
+      );
+      const signer = web3.eth.accounts.privateKeyToAccount(
+        deploymentConfig.privateKey
+      );
+      web3.eth.accounts.wallet.add(signer);
+      const estimatedGas = await web3.eth.estimateGas({
+        from: signer.address,
+        to: accountAddress,
+        value: web3.utils.toWei(`${MIN_BALANCE}`, "ether"),
+      });
+
+      const tx = await web3.eth.sendTransaction({
+        from: signer.address,
+        to: accountAddress,
+        gas: estimatedGas * deploymentConfig.gasMultiplier,
+        value: web3.utils.toWei(`${MIN_BALANCE}`, "ether"),
+      });
+
+      console.log(
+        `Topped up the ${accountName} address. Tx: `,
+        tx.transactionHash
+      );
+    }
+  }
+}
```

Comment:
can you extract this common functionality to common.ts and use it in both entropy and here?